libkbfs/backpressure_disk_limiter.go

```diff
@@ -472,101 +797,87 @@ func (bdl *backpressureDiskLimiter) getDelayLocked(
 	return time.Duration(delayScale * float64(maxDelay))
 }
 
-func (bdl *backpressureDiskLimiter) updateFreeLocked() (
-	freeBytes, freeFiles int64, err error) {
-	// Call this under lock to avoid problems with its
-	// return values going stale while blocking on
-	// bdl.lock.
-	freeBytes, freeFiles, err = bdl.freeBytesAndFilesFn()
-	if err != nil {
-		return 0, 0, err
-	}
-
-	bdl.journalFileTracker.updateFree(freeFiles)
-	bdl.journalByteTracker.updateFree(freeBytes + bdl.diskCacheByteTracker.used)
-	bdl.diskCacheByteTracker.updateFree(freeBytes + bdl.journalByteTracker.used)
-	return freeBytes, freeFiles, nil
+func (bdl *backpressureDiskLimiter) onBeforeBlockPutError(err error) (
+	availableBytes, availableFiles int64, _ error) {
+	availableBytes, availableFiles =
+		bdl.journalTracker.getSemaphoreCounts()
+	return availableBytes, availableFiles, err
 }
 
 func (bdl *backpressureDiskLimiter) beforeBlockPut(
 	ctx context.Context, blockBytes, blockFiles int64) (
 	availableBytes, availableFiles int64, err error) {
 	if blockBytes == 0 {
 		// Better to return an error than to panic in Acquire.
-		return bdl.journalByteTracker.semaphore.Count(),
-			bdl.journalFileTracker.semaphore.Count(), errors.New(
-				"backpressureDiskLimiter.beforeBlockPut called with 0 blockBytes")
+		return bdl.onBeforeBlockPutError(errors.New(
+			"backpressureDiskLimiter.beforeBlockPut called with 0 blockBytes"))
 	}
 	if blockFiles == 0 {
 		// Better to return an error than to panic in Acquire.
-		return bdl.journalByteTracker.semaphore.Count(),
-			bdl.journalFileTracker.semaphore.Count(), errors.New(
-				"backpressureDiskLimiter.beforeBlockPut called with 0 blockFiles")
+		return bdl.onBeforeBlockPutError(errors.New(
+			"backpressureDiskLimiter.beforeBlockPut called with 0 blockFiles"))
 	}
 
 	delay, err := func() (time.Duration, error) {
 		bdl.lock.Lock()
 		defer bdl.lock.Unlock()
 
-		freeBytes, freeFiles, err := bdl.updateFreeLocked()
```

Comment:
That is, I don't update the `diskCacheTracker` here, since it's not needed, and I don't update `journalTracker` in the other call site, since it's not needed there.